Remove dead / internal links and update docs

victoria-miltcheva opened the pull request:
This PR transfers over some docs from the internal detect-secrets Wiki which should be public knowledge. If I missed any docs please, let me know!

It also cleans up all dead / internal links.

justineyster commented on docs/about-detect-secrets.md:
```diff
@@ -0,0 +1,146 @@
+
+### How does this affect the current detect-secrets app?
+
+This does _not_ affect the current Github app in any way. Both services follow different flows and can be run concurrently without any issues.
```
The app is internal and has been sunset. We can remove this section.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets-stream/pull/15/commits/654bc25b26de0bc4f65030cfbe5d8fc1cea837b3

justineyster commented on docs/developer-tool-faq.md:
```diff
@@ -0,0 +1,370 @@
+# FAQs (Detect Secrets Suite - Developer tool)
```
Why move the dev tool docs into the server side tool repo? 🤔

victoria-miltcheva replied:
They were linked to from this repo's README, but it may make more sense to put them in https://github.com/IBM/detect-secrets. I'm going to go ahead and remove them from here and create a separate PR for that.

justineyster replied:
https://github.com/IBM/detect-secrets is a direct clone of the internal whitewater-detect-secrets repo. We don't push to it directly. If you want to move wiki material into source code, I'd put it in the internal one first! The public version will be updated to match automatically on deployment.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets/pull/8

victoria-miltcheva replied:
Gotcha, I wasn't aware of that. I'll go ahead and put it in the internal one. I may have created a PR against https://github.com/IBM/detect-secrets but I'll go ahead and close that 😅 . Didn't see your comment until now.

justineyster reviewed (commented):
Two things so far...

justineyster commented on README.md:
```diff
@@ -2,16 +2,16 @@
+Detect secrets stream is the server tool which ingests metadata of all (public repo by default, private repo opt-in) git pushes which happen on your company's github server. For each push, it [scans](./docs/about-detect-secrets.md#what-file-content-do-you-scan) the push content for secrets. Once found and verified, secrets will be stored in db (meta data) and SoS vault (raw secret).
 
-The live secrets are reported to Vulnerability Management team (VMT). Currently the reporting is through csv file uploaded in shared box folder. We are about to change to [report through read-only DB table](https://github.ibm.com/git-defenders/detect-secrets-discuss/issues/310).
+The live secrets are reported to the Vulnerability Management team (VMT). Currently, the reporting is through csv file uploaded in shared box folder. We are about to make a change to report through a read-only DB table.
```
This line can be removed or edited to be non-IBM-specific.

justineyster replied:
Also the read-only DB table is implemented now.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets-stream/pull/15/commits/6de0130a83c72755628551ed549d3a363ca49f86

justineyster commented on docs/about-detect-secrets.md:
```diff
@@ -0,0 +1,142 @@
+The different type of scans are:
+
+- Delta (diff) scan: scan the delta piece of any file modified or added by a commit. Suppose you have a file containing 1000 lines and your edits one line in it, it will only scan several lines surrounding the modified line. This is the behavior for detect-secrets-suite pre-receive scan.
+- Shallow scan: scan all (non-binary) files at current commit. This is the default behavior for detect-secrets-suite developer tool.
```
You could link to the public GitHub dev tool here.

justineyster commented on docs/about-detect-secrets.md:
```diff
@@ -0,0 +1,142 @@
+
+### What file content do you scan?
+
+detect-secrets-suite server runs delta (diff) scan for each commit included in the git push. If many commits included, it will scan up to 250 commits per branch. For example, if your push only contains 4 commits, then all these 4 would be scanned. In another example, if you make a new branch of master which includes thousands of existing commits, then pushed the new branch, then 250 recent commit ...
```
I believe the 250 commit limit is inaccurate based on [this line of code](https://github.com/IBM/detect-secrets-stream/blob/394cdd73c991d4627e8b14dd3d6e2597634f1067/detect_secrets_stream/gd_ingest/api.py#L61). 250 is in fact the default value for max commits as [defined in the class's init function](https://github.com/IBM/detect-secrets-stream/blob/394cdd73c991d4627e8b14dd3d6e2597634f1067/detect_secrets_stream/gd_ingest/commitparser.py#L13), but we're overriding it with 25 when the class is initialized. We may have updated the functionality since originally writing the docs.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets-stream/pull/15/commits/a19404d1aaf7642945136a424920519159e8a433

justineyster reviewed (commented):
A few suggestions!

Two more kind of general thoughts: there's a lot of use of the words "our", "we", etc. in these docs. I wonder if we should avoid those words and just talk about the tool's behavior instead of our own design decisions, now that DSS will not be managed by just our company/team. i.e. "detect-secrets-stream does xyz" instead of "we do xyz".

Secondly, the intended audience for `docs/about-detect-secrets.md` is primarily GHE end users whereas for the rest of them it's the developers managing the tool. You could maybe try to make that more clear or call that out now that these docs are living in the same place.

victoria-miltcheva commented:
>there's a lot of use of the words "our", "we", etc. in these docs. I wonder if we should avoid those words and just talk about the tool's behavior instead of our own design decisions, now that DSS will not be managed by just our company/team. i.e. "detect-secrets-stream does xyz" instead of "we do xyz".

@justineyster that's a good observation, it makes sense to refer to the tool itself rather than "we". I'll go ahead and make this change. I'll also noticed some grammar errors in the doc, so I'll fix those as well.

>Secondly, the intended audience for docs/about-detect-secrets.md is primarily GHE end users whereas for the rest of them it's the developers managing the tool. You could maybe try to make the former more clear now that these docs are living in the same place.

Will do.

victoria-miltcheva commented:
@justineyster I appreciate your feedback, and I wanted to let you know that this PR is ready for another review. These are the changes I've made:
- Fixed grammar errors
- Renamed `about-detect-secrets.md` to `detect-secrets-suite-faq.md`
- Noted the intended audience for the Detect Secrets Suite FAQ
- Removed references to "we", "our", etc.

justineyster commented on detect_secrets_stream/gd_revoker/usage.md:
```diff
@@ -2,7 +2,9 @@
-`curl -X POST https://gd-revoker.whitewater.ibm.com/api/v1/token/<uuid>/verify`
+>Note: replace `<host_suffix>` with your company-specific `gd-revoker` host suffix noted in the `ingress.gd_revoker.yaml` file.
+
+`curl -X POST https://gd-revoker.<host_suffix>./api/v1/token/<uuid>/verify`
```
The period after `<host_suffix>` should be removed.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets-stream/pull/15/commits/db95d2ecee4ece50dc5b922b4774d936c5e6132a

justineyster reviewed (approved):
One last thing. Approving so you can feel free to merge it after addressing! Thanks for working on this! 😄

killuazhu commented on README.md:
```diff
@@ -2,16 +2,14 @@
 ## Description
 
-Detect secrets stream is the server tool which ingests meta data of all (public repo by default, private repo opt-in) git pushes happened on github.ibm.com. For each push, it [scans](https://github.ibm.com/Whitewater/whitewater-detect-secrets/wiki/About-Detect-Secrets#what-file-content-do-you-scan) the push content for secrets. Once found and verified, secrets will be stored in db (meta data) and ...
+Detect Secrets Stream is a server tool which ingests metadata of all (public repositories by default, private repositories are opt-in only) git pushes on your company's GitHub server. For each push, it [scans](./docs/detect-secrets-suite-faq.md#what-file-content-do-you-scan) the push contents for secrets. Once found and verified, secrets metadata will be stored in a database, and the raw secret w ...
```
```suggestion
Detect Secrets Stream is a server tool which ingests metadata of all (public repositories by default, private repositories are opt-in only) git pushes on your company's GitHub Enterprise server. For each push, it [scans](./docs/detect-secrets-suite-faq.md#what-file-content-do-you-scan) the push contents for secrets. Once found and verified, secrets metadata will be stored in a database, and the raw secret will be stored in Vault.
```

killuazhu commented on README.md:
```diff
@@ -111,15 +110,15 @@ make test-unit-pi_cleaner
+This repo has provided a utility module which enables an admin to do many routine tasks. The utility is invoked though `python -m detect_secrets_stream.util.secret_util`
 
-Running the utility requires several environment variables. You would first unlock secrets with `git-crypt` (secret management section above), then export required environment variables. One example env var file `.env.example` has been prepared to help you.
+Running the utility requires several environment variables. You will first need to unlock secrets using `git-crypt`, then export the required environment variables. An example environment variable file (`.env.example`) has been prepared to help you.
```
```suggestion
Running the utility requires several environment variables. An example environment variable file (`.env.example`) has been provided.
```

killuazhu commented on README.md:
````diff
@@ -111,15 +110,15 @@ make test-unit-pi_cleaner
+From a `fish` shell, you can do something like below
 
 ```shell
 cp .env.example .env.prod
````
There is a better way to expose env var with [`direnv`](https://direnv.net/)   That can be added in future PR.

killuazhu commented on detect_secrets_stream/gd_revoker/usage.md:
```diff
@@ -2,7 +2,9 @@
-`curl -X POST https://gd-revoker.whitewater.ibm.com/api/v1/token/<uuid>/verify`
+>Note: replace `<host_suffix>` with your company-specific `gd-revoker` host suffix noted in the `ingress.gd_revoker.yaml` file.
+
+`curl -X POST https://gd-revoker.<host_suffix>./api/v1/token/<uuid>/verify`
```
Avoid dictating the revoker endpoint needs to start with `gd-revoker`

```suggestion
`curl -X POST https://<server>/api/v1/token/<uuid>/verify`
```

killuazhu commented on detect_secrets_stream/gd_revoker/usage.md:
```diff
@@ -15,7 +17,9 @@ Response content type is `application/json`.
-`curl -X POST https://gd-revoker.whitewater.ibm.com/api/v1/token/<uuid>/revoke`
+>Note: replace `<host_suffix>` with your company-specific `gd-revoker` host suffix noted in the `ingress.gd_revoker.yaml` file.
+
+`curl -X POST https://gd-revoker.<host_suffix>/api/v1/token/<uuid>/revoke`
```
Same as above, avoid dictating the revoker endpoint needs to start with `gd-revoker`

```suggestion
`curl -X POST https://<server>/api/v1/token/<uuid>/revoke`
```

killuazhu commented on detect_secrets_stream/secret_corpus_db/gd_db_tools.py:
```diff
@@ -118,8 +118,7 @@ def connect_db():
         db_conf = ConfUtil.load_db_conf()
-        # Creds from:
-        # https://cloud.ibm.com/services/databases-for-postgresql/crn:v1:bluemix:public:databases-for-postgresql:us-east:a%2f26bb005a5183cf92d5694dd5e93c03c2:9f5cefd3-2130-430b-8e28-bee3a0f8e105::?paneId=credentials
+        # Credentials from the PostgreSQL database
```
Good catch!

victoria-miltcheva replied:
Thanks!

killuazhu commented on docs/detect-secrets-suite-faq.md:
```diff
@@ -0,0 +1,141 @@
+
+There should be minimal delay to your push introduced by triggering the `detect-secrets` asynchronous scan. If this is not the case for you, please open a GitHub issue.
+
+### My push was made but my CI / CD didn't trigger or execute?
+
+Please take care to distinguish between the `detect-secrets-suite` service and other CI/CD.
+e.g.:
+
+- Your organization / repository's CI/CD
+- CI/CD secret checking
+- GitHub Apps
+- Pre-commit hooks you have installed locally
+
+For your own CI/CD issues, please refer to your GitHub Organization owner.
```
Let's take out this one, does not seem too relevant.

victoria-miltcheva replied:
https://github.com/IBM/detect-secrets-stream/pull/15/commits/db95d2ecee4ece50dc5b922b4774d936c5e6132a

killuazhu commented on docs/detect-secrets-suite-faq.md:
```diff
@@ -0,0 +1,141 @@
+# Welcome to `detect-secrets-suite`!
+
+This documentation is intended for GitHub Enterprise end users. The rest of the documentation in this repository is intended for developers managing the Detect Secrets Stream tool.
```
```suggestion
# Detect Secrets Stream FAQ

This documentation is intended for GitHub Enterprise end users.
```

killuazhu reviewed (changes requested):
@victoria-miltcheva thanks for making the doc updates! It looks quite nice, I've added some wording suggestions.

killuazhu reviewed (approved):
📝 🎉
